### PR TITLE
Automatic baseimage creation and deployment

### PR DESCRIPTION
**Pull request by lpramuk**
Two new tasks to be imployed by future jenkins job: 
- ```create_baseimage()```
- ```deploy_baseimage()```

### CODE REVIEW

**Comment by ntkathole** [comment]
@lpramuk If read the code correctly, the created base image wont have DNS entry of IDM server. Can we have step to update /etc/resolv.conf in `create_baseimage()` function else we need to again update image manually? Also same for /etc/ntp.conf. Let me know if I am missing something here.

File: automation_tools/baseimage.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                match_comp.group(3)
+        else:
+            image = 'unknown-{}'.format(str(time.time()).split('.')[0])
+
+    put('misc/base_image_creation/ks_rhel{}_template'.format(os_ver), 'ks.cfg')
+    run('sed -i "s|OS_URL|{}|g" ks.cfg'.format(os_url))
+    run('sed -i "s|ENCRYPT_PASS|$1$xyz$7xHVh4/yhE6P00NIXbWZA/|g" ks.cfg')
+    run('sed -i "s|AUTH_KEYS_URL|{}|g" ks.cfg'.format(os.getenv('AUTH_KEYS_URL')))
+
+    run('virt-install --connect qemu:///system -n {img} -l {url} -w bridge:br0 '
```
Comment (rplevka) [inline automation_tools/baseimage.py]: do we need to use the `bridge:br0` here? if one would liek to run this on his local or cusotm libvirt, he probalby won't have such bridge configured. + i don't think we need it there. 

not a blocker

Comment (rplevka) [reply automation_tools/baseimage.py]: don't we want to include the qemu-guest-agent serial interface here? or shall we leave this on snap-guest?

**Comment by rplevka** [comment]
@ntkathole - I would not hardcode the environment-specific configuration. Perhaps we might make the task to accept parameters tho

**Comment by lpramuk** [comment]
Yeah I know about DNS, I thought it would be Step2... But I changed my mind and incorporated in PR
DNS parametrization needs change in ks templates to works, but won't fail now.

So, changes: 
- added more parametrization of base image
- create all-in-one func ```deploy_baseimage_by _url()``` combining create and deploy together as in shell you cannot consume func return
- detect baseimage name code moved to separate func ```detect_imagename()```

File: automation_tools/baseimage.py (hunk 1 above)
Comment (lpramuk) [reply automation_tools/baseimage.py]: The latter. It has to be left for snap-guest, here we are creating only baseimage that has to have qemu-guest-agent installed (we do it already). Therefore serial interface has to added to cloned VMs and not to baseimage VM. (but could be for nothing, no loss, no gain)

**Pull request merged**